src/nrrd/formatNRRD.c

```diff
@@ -522,7 +522,11 @@ _nrrdFormatNRRD_read(FILE *file, Nrrd *nrrd, NrrdIoState *nio) {
   /* NOTE: you have to open dataFile even in the case of skipData, because
      caller might have set keepNrrdDataFileOpen, in which case you need to
      do any line or byte skipping if it is specified */
-  valsPerPiece = nrrdElementNumber(nrrd)/_nrrdDataFNNumber(nio);
+  if (nio->chunckElementNumber != 0) {
+    valsPerPiece = nio->chunckElementNumber;
```

Comment:
It seems that nrrd supports reading data from a list of data files (not just one). Shouldn't we take that into account?

Reply:
you are right, my implementation will be applied only if it is one file. I thought that in the case of multiple files, the streaming will not be necessary. We can add the streaming also in the case of multiple files, but I have not a such dataset to check the implementation. Please let me know if you have a such dataset

Reply:
I don't have such data set. If you define a chunkStartElement variable for setting the reading start location (instead of messing with the low-level byteSkip value) then it should be possible to use the current logic that iterates through files.